src/hotspot/share/gc/g1/c2/g1BarrierSetC2.cpp

```diff
@@ -658,6 +665,39 @@ bool G1BarrierSetC2::is_gc_barrier_node(Node* node) const {
 
 void G1BarrierSetC2::eliminate_gc_barrier(PhaseMacroExpand* macro, Node* node) const {
   assert(node->Opcode() == Op_CastP2X, "ConvP2XNode required");
+
```

Comment:
does G1BarrierSetC2::step_over_gc_barrier need to change?

Reply:
doesn't need to change